### ISSUE TITLE
Feature request: Add possibility to enable custom DNS server only when connected to specific WiFi SSIDs

### ISSUE THREAD

**Issue by Mattrazol** [mullvad/mullvadvpn-app#2700]
It would be great if it would be possible to have the possibility to enable/disable "Use custom DNS" setting automatically when connected to specific WiFi SSIDs. Even better could be to have a map SSID -> DNS and have a custom DNS setting associated to each specific SSID. This could trigger automatically the enable of the setting only when connected to those specific SSIDs and with a custom setting for each one of them.
I hope that it makes sense.
Thanks for your support and for the amazing service that y'all provide 😀

**Comment by faern** [comment]
Thanks for providing some feedback. Could you outline what the practical use case for this would be? Who would need it in what situation?

We have been restrictive towards introducing anything that is per network/SSID because it's easy to spoof. It would not be very hard for someone to create a fake hotspot and lure you onto it making the device think it's on your home network. If it then enables selectively enables some settings that are less privacy preserving that could be really bad. This has been explained here https://github.com/mullvad/mullvadvpn-app/issues/1884#issuecomment-654255046

**Comment by Mattrazol** [comment]
@faern My case of use is to avoid having to manually enable/disable the option when I connect to the WiFi at home. Since I have a pihole as my local DNS server I was thinking about this "smart" solution to solve that problem. But yeah, I can see the security concerns that you're raising. So u don't know what is possible to do to facilitate this process. Maybe a Taker (or IFTT) integration? Just trying to brainstorm here 😀

**Comment by pinkisemils** [comment]
If this is Linux, you could have [NetworkManager](https://developer.gnome.org/NetworkManager/stable/NetworkManager.html#refsect1) run a bash script that changes the settings when connected to a specific SSID. I'm sure there are ways to do this on other desktop OSes. But all the same security issues as above hold. You could also run a service on your Raspberry Pi that you'd use for network authentication if the security issues are a concern for you.

**Comment by Mattrazol** [comment]
@pinkisemils Sorry I forgot to mention that I'm trying to solve this on Android. I've updated the original message. Thanks for the clues though!

**Comment by pinkisemils** [comment]
The proposed solution will not work at all on Android as we don't expose a way for other applications to control our daemon. :/

**Comment by 8227846265** [comment]
I don't think this will work as well. It would be much easier to set up a Pi-Hole in the cloud and use that as your In-App Custom DNS or Android's Private DNS, that way wherever you go. 

Your Pi-Hole is connected with you.
